effectful/handlers/llm/providers.py

```diff
@@ -92,36 +94,79 @@ def _(values: Sequence) -> OpenAIMessageContent:
         return [{"type": "text", "text": str(values)}]
 
 
+@format_value.register(types.FunctionType)  # type: ignore
+def _(obj: types.FunctionType) -> OpenAIMessageContent:
+    try:
+        source = textwrap.dedent(inspect.getsource(obj)).strip()
+    except (OSError, TypeError):
+        doc = obj.__doc__ or "No docstring"
```

Comment:
We might want to raise a warning when a function-valued argument being formatted does not have a docstring.